src/handlers/next/process-protected-handler.ts

```diff
@@ -49,6 +49,8 @@ type ProcessAsyncSaleorProtectedHandler = (
 /**
  * Perform security checks on given request and return ProtectedHandlerContext object.
  * In case of validation issues, instance of the ProtectedHandlerError will be thrown.
+ *
+ * Can pass entire next request or Headers with saleorApiUrl and token
```

Comment:
Can it? Because the type only picks the `headers` key:

```ts
Pick<NextApiRequest, "headers">;
```

Reply:
Yes, Pick is a subset, so everything that contains `{headers: Headers}` will pass.

Reply:
I am not sure I understand. I created [a ts playground](https://www.typescriptlang.org/play?#code/C4TwDgpgBAShCOUC8UDeAoKWoAsIEMATCAJwGcAuKM4EgSwDsBzAbk2wDMB7Lqm+5mwC+6dKEhQAKhBrIoABToBjANYAeOPAA0UAER4ipMroB8opVwaz8VabJQZsUAEb4SVffgA2XXVvZYBsTkHn7oQkA) to explain what I mean.

Reply:
This is how it works

https://www.typescriptlang.org/play?#code/C4TwDgpgBAShCOUC8UDeAoKWoAsIEMATCAJwGcAuKM4EgSwDsBzAbk2wDMB7Lqm+5mwC+6dKEhQAKhBrIoABToBjANYAeOPAA0UAER4ipMroB8opVwaz8SpRDDBpslAApgM4FSfAAlGhHoFlbAUCQIVJpyGNi4BMTkVADk7jSJWuxY3LxQiVmJ6AE2dg7eLmHwPkA

Reply:
Ah, I get it! Thanks for the explanation.